pkgs/test/haskell-setBuildTarget/default.nix

```diff
@@ -0,0 +1,26 @@
+{ pkgs, haskellPackages }: with pkgs.haskell.lib;
+
+
+let
+  drv     = haskellPackages.callCabal2nix "haskell-setBuildTarget" ./. {};
```

Comment:
Ah, sorry I didn't catch this earlier.  `callCabal2nix` actually won't work here because it does IFD.  This won't work on Hydra.

I think ofborg catches this, since evaluation still fails (https://gist.github.com/GrahamcOfBorg/4c5ca9d702aead273a0b8cedf6cb9fe1), but from the error message it is not clear what is going on.

Reply:
The error message is an import failure of a store path, so that's precisely the ifd failure :)

Reply:
@cdepillabout I admit I am dependant on `callCabal2nix` and don't know how to do what you ask.

Reply:
`callCabal2nix` just runs `cabal2nix` and uses imports the resulting `.nix` source file. You can just generate that file once by running the program manually.

Reply:
@Fresheyeball I added commit a77475b that shows what I was trying to say.

If you take a look at this and agree this looks alright, I think this should be okay to be merged in.